### PR TITLE
Game Class - Factory type object

### PR DESCRIPTION
**Pull request by IsaacSShade**
Sets up every city, connects them, and creates all the cures. Passes both of these lists to the gameboard to initialize it, when given a list of players. 

I don't know if this class is 100% complete, but for the GUI people I'm getting this out there ASAP.

Merging with the Gameboard branch because it's all based on the Gameboard constructor.

Notes: Izzy, you should add to the constructor a difficulty level, and I'm thinking we should probably move the construction of the deck objects up to this object because the Game object's purpose is to construct things for the Gameboard. Let me know what you think!

### CODE REVIEW

**Review by tonixsmm — APPROVED** [review]
Thanks Aiden! It looks good.

**Review by kalfaro2025 — APPROVED** [review]
Looks great!

File: src/main/java/edu/gonzaga/Game.java [hunk 1]
```diff
@@ -1,5 +1,169 @@
...
+        City sanFrancisco = createCity(cityList, Color.BLUE, "San Francisco", chicago, null, null);
+        City montreal = createCity(cityList, Color.BLUE, "Montreal", chicago, null, null);
+        City newYork = createCity(cityList, Color.BLUE, "New York", montreal, null, null);
+        City washington = createCity(cityList, Color.BLUE, "Washington", montreal, newYork, atlanta);
+        City london = createCity(cityList, Color.BLUE, "London", newYork, null, null);
+        City madrid = createCity(cityList, Color.BLUE, "Madrid", newYork, london, null);
+        City paris = createCity(cityList, Color.BLUE, "Paris", madrid, london, null);
+        City essen = createCity(cityList, Color.BLUE, "Essen", paris, london, null);
+        City milan = createCity(cityList, Color.BLUE, "Milan", paris, essen, null);
+        City sainPetersburg = createCity(cityList, Color.BLUE, "Saint Petersburg", essen, null, null);
```
Comment (izzytilles) [inline src/main/java/edu/gonzaga/Game.java:111]: not big deal, but type onn variable name

Comment (izzytilles) [reply src/main/java/edu/gonzaga/Game.java:111]: wow that is so embarrassing that I had a typo in that comment

**Review by izzytilles — APPROVED** [review]
It looks very solid! I agree with you about moving the Deck initialization to Game since it initializes everything for the Gameboard. I had conceptualized the Game class as the class in which we would set up the gameplay flow and logic, so I think we need to make another class to handle that. Let me know what you are thinking on that front!

**Review by kalfaro2025 — APPROVED** [review]

**Review by izzytilles — APPROVED** [review]
This all looks great. Nice work!

**Pull request merged**